### DISCUSSION TITLE
Merge PDFs

### DISCUSSION THREAD

**Discussion by Parag411** [nextgenhealthcare/connect#5434] [General]
Hello Guys,

I am generating multiple PDFs as an attachment as well as a pdf file from a XML file.
At the end, I want to merge this PDFs into 1 PDF.
I tried to merge base64 code into 1 but it didn't worked.
I tried to merge the actual pdf files as per [post](https://forums.mirthproject.io/forum/mirth-connect/support/18564-merge-pdfs) but it didn't worked.

Could you please suggest the ideal way to do this ?
Mirth version: 4.0.1

Any help is appreciated.

Thank you.

**Comment by rjkroll** [comment]
The post has the correct solution.     You can test this yourself by creating a simple Javascript Reader channel and putting this in a transformer step:


```
//existing file
var pdf1 = FileUtil.readBytes('E:/test/a.pdf');
//file to be merged
var pdf2 = FileUtil.readBytes('E:/test/b.pdf');
var merger = new org.apache.pdfbox.util.PDFMergerUtility();
   merger.addSource(new java.io.ByteArrayInputStream(pdf1));
   merger.addSource(new java.io.ByteArrayInputStream(pdf2));
   // do the rest with the merger
//destination
var bout2 = new java.io.BufferedOutputStream(new java.io.FileOutputStream('E:/test/c.pdf'));
//magic
            merger.setDestinationStream(bout2);
            merger.mergeDocuments();
```

**Reply by Parag411** [reply]
> The post has the correct solution. You can test this yourself by creating a simple Javascript Reader channel and putting this in a transformer step:
> 
> ```
> //existing file
> var pdf1 = FileUtil.readBytes('E:/test/a.pdf');
> //file to be merged
> var pdf2 = FileUtil.readBytes('E:/test/b.pdf');
> var merger = new org.apache.pdfbox.util.PDFMergerUtility();
>    merger.addSource(new java.io.ByteArrayInputStream(pdf1));
>    merger.addSource(new java.io.ByteArrayInputStream(pdf2));
>    // do the rest with the merger
> //destination
> var bout2 = new java.io.BufferedOutputStream(new java.io.FileOutputStream('E:/test/c.pdf'));
> //magic
>             merger.setDestinationStream(bout2);
>             merger.mergeDocuments();
> ```

This post talks about merging actual PDF files.
I want to merge the PDF files using base64 code to avoid writing them to a local drive and then merge them.
Could you please let me know how to merge the PDFs when Mirth generates them in Document Writer connector as an attachment (Not actual PDF file)?

Thanks

**Comment by pacmano1** [comment]
OP is cross posting, see forums discussion at https://forums.mirthproject.io/forum/mirth-connect/support/17980-merge-2-base64-encoded-pdf-strings#post182222

**Reply by pacmano1** [reply]
![image](https://user-images.githubusercontent.com/44065187/193975554-047897db-9c65-4bf4-8505-6e3c515a86bc.png)

**Reply by pacmano1** [reply]
Answered above for mirth v4.0..1
